servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerBinder.java

```diff
@@ -183,7 +184,13 @@ protected void initChannel(final Channel channel) {
         return new SubscribableSingle<ServerContext>() {
             @Override
             protected void handleSubscribe(Subscriber<? super ServerContext> subscriber) {
-                subscriber.onSubscribe(() -> future.cancel(true));
+                try {
+                    subscriber.onSubscribe(() -> future.cancel(true));
+                } catch (Throwable t) {
+                    handleExceptionFromOnSubscribe(subscriber, t);
+                    future.cancel(true);
```

Comment:
I don't know if this is enough: I think we also need to add a listener to the future in the case that cancel isn't effective to cleanup the resource once it resolves.

Reply:
Great catch!
Added: https://github.com/apple/servicetalk/pull/3046/commits/ca099b70896305b1b7b381c18c95a9837146236e